pkg/kubectl/cmd/set/set_image.go

```diff
@@ -164,12 +166,22 @@ func (o *ImageOptions) Run() error {
 		transformed := false
 		_, err := o.UpdatePodSpecForObject(info.Object, func(spec *api.PodSpec) error {
 			for name, image := range o.ContainerImages {
-				containerFound := false
+				var (
+					containerFound bool
+					err            error
+					resolved       string
+				)
 				// Find the container to update, and update its image
 				for i, c := range spec.Containers {
 					if c.Name == name || name == "*" {
-						spec.Containers[i].Image = image
 						containerFound = true
+						if len(resolved) == 0 {
+							if resolved, err = o.ResolveImage(image); err != nil {
+								allErrs = append(allErrs, fmt.Errorf("error: unable to resolve image %q for container %q: %v", image, name, err))
```

Comment:
If there's error returned in '*' case (`len(resolved)` will still be `0`) we'll loop again and append another error for the same image? Seems redundant. Why not move this call to `ResolveImage` outside of this spec.Containers for loop?